### ISSUE TITLE
Error 

### ISSUE THREAD

**Issue by WhiteSide82** [B16f00t/whapa#106]
I update to the lastest version. work well with one account, but change to another and present this error.

![image](https://user-images.githubusercontent.com/85359384/120829732-5a090580-c523-11eb-95e4-a2e25d8ece2d.png)

**Comment by Gas30111971** [comment]
It used to work open accounts.google.com/displayunlockcaptcha within the account and unlock, but since two days ago, it seems not being working, the message keep showing again and again. Does any one know how to resolve the issue?
Btw, it happens with earlier versions, so probably the problem is in google, not in the app
Thanks

**Comment by Pinguino71** [comment]
You may need to update python with the requeriment.txt file provided in the solution

pip install --upgrade -r ./doc/requirements.txt

It solved the issue in my case

**Comment by faouez15** [comment]
Still not working for me

**Comment by TehBotolSosro** [comment]
it used to work fine, but now same issue as above. 
already tried upgrading the python to the newest version and already did the requirement.txt and yet still not working.

**Comment by B16f00t** [comment]
I would like you to test if this script works for you, https://github.com/YuriCosta/WhatsApp-GD-Extractor-Multithread and if it does, let me know so I can incorporate it in my tool.
Whapa currently works with some accounts and not others, I'm not sure if it's the python version, Whatsapp, or something from the google account.

**Comment by sspacenerd** [comment]
I tried a couple of days ago the script of Yuricosta, and still doesnt work for me! It gives me the same issue (Needs browser and i dont have any 2AF activated and i do the Display unlock captcha!)

**Comment by TehBotolSosro** [comment]
Thank you the new script worked great

using gpsoauth==0.4.3 result in {'Error': 'BadAuthentication'}
using gpsoauth== 1.0.0 result in {'Error': 'NeedsBrowser', 'Url': 
after doing https://accounts.google.com/b/0/DisplayUnlockCaptcha
then it worked fine like whapa has few months ago

small observation in the new script seem need to be improved since doing Info result in KeyError: 'passwordProtectedBackupEnabled'
also got an error the first time it sync (downloading he backup files) but resolved it by repeating the sync option
```
    self.get(file["name"].replace("%", "%25").replace("+", "%2B"), {"alt": "media"}, stream=True)
  File "C:\NewWhapa\WhatsAppGDExtract.py", line 75, in get
    response = requests.get(
  File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

> 
> I would like you to test if this script works for you, https://github.com/YuriCosta/WhatsApp-GD-Extractor-Multithread and if it does, let me know so I can incorporate it in my tool.
> Whapa currently works with some accounts and not others, I'm not sure if it's the python version, Whatsapp, or something from the google account.

**Comment by sspacenerd** [comment]
Is there a way on using this new script without the Android ID? I think thats whats causing my problem!

**Comment by TehBotolSosro** [comment]
> 
> 
> Is there a way on using this new script without the Android ID? I think thats whats causing my problem!



> 
> 
> Is there a way on using this new script without the Android ID? I think thats whats causing my problem!

i always use same android ID for several account without issue, since without entering the android id it will cause a new android device every time you try to use the script, and Your devices on your google account will be filled with Android Unknown Device

**Comment by sspacenerd** [comment]
> > Is there a way on using this new script without the Android ID? I think thats whats causing my problem!
> 
> > Is there a way on using this new script without the Android ID? I think thats whats causing my problem!
> 
> i always use same android ID for several account without issue (even if the device are not the one that are linked to that google account), since without entering the android id it will cause a new android device every time you try to use the script, and YOUR DEVICES on your google account will be filled with Android Unknown Device


I see! I used again the script with my android ID and all i get is this: {'Error': 'BadAuthentication'}. Do you know why i get this error? Im using gpsoauth.0.4.3!

**Comment by B16f00t** [comment]
> 
> 
> Thank you the new script it worked great
> 
> using gpsoauth-0.4.3 result in {'Error': 'BadAuthentication'}
> using gpsoauth1.0.0 result in {'Error': 'NeedsBrowser', 'Url':
> after doing https://accounts.google.com/b/0/DisplayUnlockCaptcha
> then it worked fine like whapa has few months ago
> 
> also using Charles Web Proxy with ssl proxying (to see the request/response) result in {'Error': 'BadAuthentication'}, u need to turn off the ssl proxying so the script can work
> 
> small observation in the new script seem need to be improved since doing Info result in KeyError: 'passwordProtectedBackupEnabled'
> also got an error the first time it sync (downloading he backup files) but resolved it by repeating the sync option (probably a bad connection)
> 
> ```
>     self.get(file["name"].replace("%", "%25").replace("+", "%2B"), {"alt": "media"}, stream=True)
>   File "C:\NewWhapa\WhatsAppGDExtract.py", line 75, in get
>     response = requests.get(
>   File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\api.py", line 76, in get
>     return request('get', url, params=params, **kwargs)
>   File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\api.py", line 61, in request
>     return session.request(method=method, url=url, **kwargs)
>   File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\sessions.py", line 530, in request
>     resp = self.send(prep, **send_kwargs)
>   File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\sessions.py", line 643, in send
>     r = adapter.send(request, **kwargs)
>   File "C:\Users\HDMF\AppData\Local\Programs\Python\Python39-32\lib\site-packages\requests\adapters.py", line 498, in send
>     raise ConnectionError(err, request=request)
> requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
> ```
> 
> also testing it on a Google Account that have two whatsapp backup (two phone number) result only one of them that are downloaded . and it end with an error
> 
> ```
> Sync Result 
> Progress: 100.000% PTT-20210111-WA0015.opus
> 234 files (45MiB)
> Traceback (most recent call last):
>   File "C:\twowha\WhatsAppGDExtract.py", line 230, in <module>
>     main(sys.argv)
>   File "C:\twowha\WhatsAppGDExtract.py", line 224, in main
>     human_size(int(backup["sizeBytes"])),
> KeyError: 'sizeBytes'
> ```
> 
> Info Result
> 
> Traceback (most recent call last):
> File "C:\twowha\WhatsAppGDExtract.py", line 230, in
> main(sys.argv)
> File "C:\twowha\WhatsAppGDExtract.py", line 207, in main
> print(backup_info(backup))
> File "C:\twowha\WhatsAppGDExtract.py", line 179, in backup_info
> metadata = json.loads(backup["metadata"])
> KeyError: 'metadata'
> 
> ```
> 
> 
> > 
> > I would like you to test if this script works for you, https://github.com/YuriCosta/WhatsApp-GD-Extractor-Multithread and if it does, let me know so I can incorporate it in my tool.
> > Whapa currently works with some accounts and not others, I'm not sure if it's the python version, Whatsapp, or something from the google account.
> ```

dows Whapa work for you or only works the YuriCosta Script?

**Comment by B16f00t** [comment]
> 
> 
> > > Is there a way on using this new script without the Android ID? I think thats whats causing my problem!
> > 
> > 
> > > Is there a way on using this new script without the Android ID? I think thats whats causing my problem!
> > 
> > 
> > i always use same android ID for several account without issue (even if the device are not the one that are linked to that google account), since without entering the android id it will cause a new android device every time you try to use the script, and YOUR DEVICES on your google account will be filled with Android Unknown Device
> 
> I see! I used again the script with my android ID and all i get is this: {'Error': 'BadAuthentication'}. Do you know why i get this error? Im using gpsoauth.0.4.3!

Update gpsoauth to the latest version

**Comment by TehBotolSosro** [comment]
Windows Whapa used to work for me few months ago, but few days ago when i try Whapa again it keep saying Error': 'NeedsBrowser', even after doing the unlockcaptcha url (and still doesn't work even after updating the python and the requirement package)

so for now only YuriCosta Script that work for me, Tried it with several google account, and only YuriCosta script that work

could it be the old gpsoauth (0.4.3) that are causing the error? and only v1.0.0 that work now?, sadly i tried gpsoauth with whapa but it wasn't compatible (something about signature error).

> > ```
W> 
> dows Whapa work for you or only works the YuriCosta Script?

**Comment by sspacenerd** [comment]
Hey! I managed to get it work for me, I downgrade my version of gpsoauth0.4.3 to 0.4.2 and it works pretty good! Thanks for your help c:

**Comment by TehBotolSosro** [comment]
ah yeah the YuriCosta does indeed work with gpsoauth 0.4.2  and 1.0.0 but not with 0.4.3 (Error': 'BadAuthentication)

**Comment by dilinjabass** [comment]
I've been testing the YuriCosta app and it seems to work ok. Some backups have an error on a file which closes the whole sync and you cant get the whole backup. But other backups synced fine, even from multiple numbers on the same account.  (it automatically downloads all backups on the account)
I didn't edit the android ID, I just left it all 0's in every case.

One nice thing with the YuriCosta script is when it syncs the backup, it shows you in percentage how much of the backup is downloaded, kinda useful to see how much longer it has.

**Comment by B16f00t** [comment]
Version 1.5 fix the issue, python 3.8 is required and update requirements.
